core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -104,47 +116,94 @@ public MountTable(UfsManager ufsManager, MountInfo rootMountInfo) {
    * @throws InvalidPathException if an invalid path is encountered
    */
   public void add(Supplier<JournalContext> journalContext, AlluxioURI alluxioUri, AlluxioURI ufsUri,
-      long mountId, MountPOptions options) throws FileAlreadyExistsException, InvalidPathException {
+      long mountId, MountPOptions options) throws FileAlreadyExistsException, InvalidPathException,
+      IOException {
+    try (LockResource r = new LockResource(mWriteLock)) {
+      // validate the Mount operation first, error will be thrown if the operation is invalid
+      ValidatedPathPair validatedPathPair = validateMountPoint(alluxioUri, ufsUri);
+      add(journalContext, validatedPathPair, mountId, options);
+    }
+  }
+
+  /**
+   * Mounts based on the given ValidatedPathPair. It will directly insert a new entry into
+   * MountTable via the mState. This method is NOT ThreadSafe. When calling this add, lockResource
+   * will not be applied and the
+   * caller MUST hold the write lock resource.
+   * @param journalContext the journal context
+   * @param validatedPair the validated mount entry
+   * @param mountId the mount id
+   * @param options the mount options
+   */
+  public void add(Supplier<JournalContext> journalContext,
+      ValidatedPathPair validatedPair, long mountId, MountPOptions options) {
+    AlluxioURI alluxioUri = validatedPair.getAlluxioPath();
+    AlluxioURI ufsUri = validatedPair.getUfsPath();
```

Comment:
With `ValidatedPathPair` removed, you can just pass in `AlluxioURI alluxioPath, AlluxioURI ufsPath` in the arguments